classes/Pods.php

```diff
@@ -288,16 +288,16 @@ public function __construct( $pod = null, $id = null ) {
 			if ( $queried_object ) {
 				$id_lookup = true;
 
-				if ( isset( $queried_object->post_type ) ) {
+				if ( $queried_object instanceof WP_Post ) {
 					// Post Type Singular.
 					$pod = $queried_object->post_type;
-				} elseif ( isset( $queried_object->taxonomy ) ) {
+				} elseif ( $queried_object instanceof WP_Term ) {
 					// Term Archive.
 					$pod = $queried_object->taxonomy;
-				} elseif ( isset( $queried_object->user_login ) ) {
+				} elseif ( $queried_object instanceof WP_User ) {
 					// Author Archive.
 					$pod = 'user';
-				} elseif ( isset( $queried_object->public, $queried_object->name ) ) {
+				} elseif ( $queried_object instanceof WP_Post_Type && $queried_object->public ) {
```

Comment:
```suggestion
				} elseif ( $queried_object instanceof WP_Post_Type ) {
```

We didn't previously check if it was public

Reply:
Oh right, it was only an `isset`, not `! empty`